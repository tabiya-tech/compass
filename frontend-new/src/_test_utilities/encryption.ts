import { webcrypto as crypto } from "crypto";
import { EncryptionConfig } from "src/sensitiveData/config/encryptionConfig";

// Constraining the key size to 2048 or 4096 bits for RSA as the backend expect the 4096 in calculating the maximum size of the encrypted key.
export async function generateRSACryptoPairKey(keySize: 2048 | 4096) {
  return await crypto.subtle.generateKey(
    {
      name: EncryptionConfig.RSA.ALGORITHM,
      modulusLength: keySize,
      // 3 is a common exponent in practice
      publicExponent: new Uint8Array([1, 0, 1]),
      hash: EncryptionConfig.RSA.KEY_HASH_FN,
    },
    true, // Whether the key is extractable (i.e. can be used in exportKey)
    ["encrypt", "decrypt"] // the key can be used for both encryption and decryption
  );
}

function arrayBufferToBase64(buffer: ArrayBuffer): string {
  // @ts-ignore
  const binary = String.fromCharCode(...new Uint8Array(buffer));

  return Buffer.from(binary, "binary").toString("base64");
}

/**
 * Export a CryptoKey to a PEM formatted string (public key)
 *
 * @param key The CryptoKey to export as a PEM formatted string
 * @returns The PEM formatted string.
 */
export async function exportCryptoPublicKey(key: CryptoKey) {
  // Export the key to ArrayBuffer
  const exported = await crypto.subtle.exportKey(EncryptionConfig.RSA.FORMAT, key);

  const base64String = arrayBufferToBase64(exported);

  // add the PEM header and footer to make it a PEM formatted string.
  return `-----BEGIN PUBLIC KEY-----\n${base64String}\n-----END PUBLIC KEY-----`;
}

/**
 * Decrypt the encrypted data with the private key
 * using RSA Algorithm.
 *
 * @param privateKey
 * @param encryptedData
 */
export async function decryptWithRSA(privateKey: CryptoKey, encryptedData: ArrayBuffer) {
  return await crypto.subtle.decrypt(
    {
      name: EncryptionConfig.RSA.ALGORITHM,
    },
    privateKey,
    encryptedData
  );
}

export async function decryptWithAES(encryptedData: ArrayBuffer, iv: ArrayBuffer, key: ArrayBuffer) {
  try {
    const importedKey = await crypto.subtle.importKey(
      EncryptionConfig.AES.FORMAT,
      key,
      EncryptionConfig.AES.ALGORITHM,
      true,
      ["decrypt"]
    );

    return await crypto.subtle.decrypt(
      {
        name: EncryptionConfig.AES.ALGORITHM,
        iv,
        length: EncryptionConfig.AES.KEY_LEN,
        tagLength: EncryptionConfig.AES.TAG_LEN,
      },
      importedKey,
      encryptedData
    );
  } catch (e) {
    console.error(e);
    throw e;
  }
}
