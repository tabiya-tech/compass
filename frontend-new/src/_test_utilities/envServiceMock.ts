jest.mock("src/envService", () => ({
  getFirebaseAPIKey: jest.fn(() => "mock-api-key"),
  getFirebaseDomain: jest.fn(() => "mock-auth-domain"),
  getBackendUrl: jest.fn(() => "mock-backend-url"),
  getApplicationLoginCode: jest.fn(() => ""),
  getApplicationRegistrationCode: jest.fn(() => ""),
  getLoginCodeDisabled: jest.fn(() => "false"),
  getRegistrationDisabled: jest.fn(() => "false"),
  getMetricsEnabled: jest.fn(() => "true"),
  getMetricsConfig: jest.fn(() => ""),
  getCvUploadEnabled: jest.fn(() => "true"),
  getSocialAuthDisabled: jest.fn(() => "false"),
}));
