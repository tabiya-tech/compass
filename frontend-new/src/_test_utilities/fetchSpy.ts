import { ExtendedRequestInit } from "src/utils/customFetch/customFetch";

export function setupFetchSpy(
  expectedStatus: number,
  expectedResponseBody: string | object | undefined,
  contentType: "" | "application/json;charset=UTF-8"
): jest.SpyInstance {
  const responseBody =
    typeof expectedResponseBody === "string" ? expectedResponseBody : JSON.stringify(expectedResponseBody);
  const expectedResponse = new Response(responseBody, {
    status: expectedStatus,
    headers: { "Content-Type": contentType },
  });
  return jest.spyOn(window, "fetch").mockResolvedValue(expectedResponse);
}

// spy on authenticated fetch requests
export function setupAPIServiceSpy(
  expectedStatus: number,
  expectedResponseBody: string | object | undefined,
  contentType: "" | "application/json;charset=UTF-8"
): jest.SpyInstance {
  const responseBody =
    typeof expectedResponseBody === "string" ? expectedResponseBody : JSON.stringify(expectedResponseBody);
  const expectedResponse = new Response(responseBody, {
    status: expectedStatus,
    headers: { "Content-Type": contentType },
  });

  jest.spyOn(window, "fetch").mockResolvedValue(expectedResponse);
  return jest.spyOn(require("src/utils/customFetch/customFetch"), "customFetch");
}

export function expectCorrectFetchRequest(fetchSpy: jest.SpyInstance, expectedUrl: string, expectedConfig: ExtendedRequestInit) {
  expect(fetchSpy).toHaveBeenCalledTimes(1);
  const [actualUrl, actualConfig] = fetchSpy.mock.calls[0];
  expect(actualUrl).toBe(expectedUrl);
  expect(actualConfig.method).toBe(expectedConfig.method);

  // Compare headers if they exist
  if(actualConfig.headers) {
    // Handle headers comparison - convert Headers object to plain object if needed
    const actualHeaders = actualConfig.headers instanceof Headers
      ? Object.fromEntries(actualConfig.headers.entries())
      : actualConfig.headers;
    expect(actualHeaders).toEqual(
      expect.objectContaining(expectedConfig.headers),
    );
  }

  // Compare bodies by parsing JSON if they exist
  if (actualConfig.body && expectedConfig.body) {
    let actualBody: unknown;
    let expectedBody: unknown;
    
    try {
      actualBody = JSON.parse(JSON.stringify(actualConfig.body));
      expectedBody = JSON.parse(JSON.stringify(expectedConfig.body));
    } catch (e) {
      console.error('Error parsing request bodies:', e);
      throw new Error('Failed to parse request bodies for comparison');
    }
    expect(actualBody).toEqual(expectedBody);
  }

  // Check any extra properties if provided
  type ExtendedConfigKey = keyof ExtendedRequestInit;
  const extraProperties: ExtendedConfigKey[] = [
    'expectedStatusCode',
    'serviceName',
    'serviceFunction',
    'failureMessage',
    'expectedContentType'
  ] as ExtendedConfigKey[];

  extraProperties.forEach((key) => {
    if (expectedConfig[key] !== undefined) {
      expect(actualConfig[key]).toEqual(expectedConfig[key]);
    }
  });
}

