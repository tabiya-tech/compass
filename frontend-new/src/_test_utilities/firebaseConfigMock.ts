import { AuthProviderIds } from "src/auth/auth.types";

const auth = jest.fn(() => ({
  signInWithCustomToken: jest.fn(),
  signInWithEmailAndPassword: jest.fn(),
  createUserWithEmailAndPassword: jest.fn(),
  signOut: jest.fn(),
  GoogleAuthProvider: { PROVIDER_ID: AuthProviderIds.GOOGLE },
}));

export function mockFirebaseConfig() {
  jest.spyOn(require("src/auth/firebaseConfig"), "auth").mockImplementation(auth);
}
