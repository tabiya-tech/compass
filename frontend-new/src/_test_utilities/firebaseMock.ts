import { AuthProviderIds } from "src/auth/auth.types";

const auth = {
  currentUser: {
    getIdToken: jest.fn(),
    getTokenResult: jest.fn(),
  },
  onAuthStateChanged: jest.fn().mockReturnValue(jest.fn()),
  signInWithPopup: jest.fn(),
  GoogleAuthProvider: { PROVIDER_ID: AuthProviderIds.GOOGLE },
};

const firebase = {
  initializeApp: jest.fn(),
  auth: () => auth,
};

export default firebase;
