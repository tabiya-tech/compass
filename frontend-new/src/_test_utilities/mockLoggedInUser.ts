import { AuthContextValue, TabiyaUser, authContextDefaultValue } from "src/auth/AuthProvider";

export function mockLoggedInUser(mockedValues: Partial<AuthContextValue>) {
  let user: TabiyaUser = mockedValues.user || TestUser;
  jest.spyOn(require("src/auth/hooks/useAuthUser"), "useAuthUser").mockReturnValue({
    ...authContextDefaultValue,
    user: user,
    updateUser: (user: TabiyaUser) => null,
    updateUserByAccessToken: () => {},
    ...mockedValues,
  });
}

export const TestUser = {
  id: "0001",
  name: "Test User",
  email: "test@email.com",
};
