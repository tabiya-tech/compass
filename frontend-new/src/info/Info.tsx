import React, { useEffect, useMemo, useState } from "react";
import { Box, Skeleton, Typography, useTheme } from "@mui/material";
import InfoService from "src/info/info.service";
import FeedbackButton from "src/feedback/FeedbackButton";

const uniqueId = "37d307ae-4f1e-4d8d-bafe-fd642f8af4dc";
export const DATA_TEST_ID = {
  INFO_ROOT: `version-${uniqueId}`,
  VERSION_FRONTEND_ROOT: `version-frontend-${uniqueId}`,
  VERSION_BACKEND_ROOT: `version-backend-${uniqueId}`,
};

const VersionInfoItem = ({ title, value, skeleton }: { title: string; value: string; skeleton?: boolean }) => {
  return (
    <Box>
      <Typography variant="h6">{title}</Typography>
      <Typography variant="body1">
        {skeleton ? <Skeleton>{value.replace(/./g, "\u00A0\u00A0")}</Skeleton> : value}
      </Typography>
    </Box>
  );
};

const VersionContainer = ({ dataTestId, title, info }: { dataTestId: string; title: string; info: InfoProps }) => {
  const theme = useTheme();
  return (
    <Box display="flex" gap={theme.tabiyaSpacing.xl} data-testid={dataTestId}>
      <Typography variant="h6" sx={{ minWidth: "100px" }}>
        {title}
      </Typography>
      {info ? (
        <Box display="flex" flexDirection="column" gap={theme.tabiyaSpacing.sm}>
          <VersionInfoItem title="Date" value={info.date} />
          <VersionInfoItem title="Version" value={info.branch} />
          <VersionInfoItem title="Build Number" value={info.buildNumber} />
          <VersionInfoItem title="GIT SHA" value={info.sha} />
        </Box>
      ) : (
        <Box display="flex" flexDirection="column" gap={theme.tabiyaSpacing.sm}>
          <VersionInfoItem title="Date" value={"0000-00-00T00:00:00.000Z"} skeleton={true} />
          <VersionInfoItem title="Version" value={"foo"} skeleton={true} />
          <VersionInfoItem title="Build Number" value={"000"} skeleton={true} />
          <VersionInfoItem title="GIT SHA" value={"foofoofoofoofoofoofoofoofoofoofoofoofoo"} skeleton={true} />
        </Box>
      )}
    </Box>
  );
};

export interface InfoProps {
  date: string;
  branch: string;
  buildNumber: string;
  sha: string;
}

const ApplicationInfoMain = (props: { versions: InfoProps[] }) => {
  const theme = useTheme();
  return (
    <Box display="flex" flexDirection="column" gap={theme.tabiyaSpacing.xl}>
      <VersionContainer title="Frontend" info={props.versions[0]} dataTestId={DATA_TEST_ID.VERSION_FRONTEND_ROOT} />
      <VersionContainer title="Backend" info={props.versions[1]} dataTestId={DATA_TEST_ID.VERSION_BACKEND_ROOT} />
    </Box>
  );
};

const Info = () => {
  const [versions, setVersions] = useState<InfoProps[]>([]);
  const infoService = useMemo(() => new InfoService(), []);
  useEffect(() => {
    infoService.loadInfo().then((data) => setVersions(data));
  }, [infoService]);

  return (
    <div style={{ width: "100%", height: "100%" }} data-testid={DATA_TEST_ID.INFO_ROOT}>
      <ApplicationInfoMain versions={versions} />
      <FeedbackButton bottomAlign={true} />
    </div>
  );
};

export default Info;
