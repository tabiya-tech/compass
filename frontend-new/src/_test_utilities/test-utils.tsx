// Based on https://testing-library.com/docs/react-testing-library/setup/
import React, { ReactElement, ReactNode } from "react";
import { render, renderHook, RenderHookOptions, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "@mui/material";
import applicationTheme, { ThemeMode } from "src/theme/applicationTheme/applicationTheme";

// Import the Firebase mock utilities
import "src/_test_utilities/firebaseMock";
import SnackbarProvider from "src/theme/SnackbarProvider/SnackbarProvider";
import { IsOnlineProvider } from "src/app/isOnlineProvider/IsOnlineProvider";
jest.mock("firebase/compat/app", () => {
  return {
    initializeApp: jest.fn(),
    auth: jest.fn().mockReturnValue({
      signOut: jest.fn(),
      currentUser: {
        getIdToken: jest.fn(),
        getTokenResult: jest.fn(),
      },
      onAuthStateChanged: jest.fn().mockReturnValue(jest.fn()),
      signInWithPopup: jest.fn(),
      GoogleAuthProvider: { PROVIDER_ID: "google.com" },
    }),
  };
});

jest.mock("firebaseui", () => {
  return {
    auth: {
      AuthUI: {
        start: jest.fn(),
        getInstance: jest.fn().mockReturnValue({
          start: jest.fn(),
          reset: jest.fn(),
        }),
        reset: jest.fn(),
      },
    },
  };
});

jest.mock("@sentry/react", () => ({
  withSentry: (Component: any) => Component,
  getFeedback: jest.fn(),
  withProfiler: (Component: any) => Component,
  wrapCreateBrowserRouter: (Component: any) => Component,
}));

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <IsOnlineProvider>
      <ThemeProvider theme={applicationTheme(ThemeMode.LIGHT)}>
        <SnackbarProvider>{children}</SnackbarProvider>
      </ThemeProvider>
    </IsOnlineProvider>
  );
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, "wrapper">) =>
  render(ui, { wrapper: AllTheProviders, ...options });

function customRenderHook<T>(
  hook: () => T,
  options?: Omit<RenderHookOptions<T>, "wrapper"> & { wrapper?: React.ComponentType }
) {
  const CombinedWrapper = ({ children }: { children: ReactNode }) => {
    const Wrapper = options?.wrapper ?? React.Fragment;
    return (
      <AllTheProviders>
        <Wrapper>{children}</Wrapper>
      </AllTheProviders>
    );
  };

  return renderHook(hook, { wrapper: CombinedWrapper, ...options });
}

export * from "@testing-library/react";
export { customRender as render };
export { customRenderHook as renderHook };
