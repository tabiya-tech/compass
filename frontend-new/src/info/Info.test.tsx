// mute the console
import "src/_test_utilities/consoleMock";

import Info, { DATA_TEST_ID, InfoProps } from "./Info";
import { act, render, screen } from "src/_test_utilities/test-utils";
import InfoService from "./info.service";
import { HashRouter } from "react-router-dom";

// Mock the info service
jest.mock("./info.service", () => {
  const mockInfoService = jest.fn();
  mockInfoService.prototype.loadInfo = jest.fn().mockImplementation(() => {
    return Promise.resolve([]);
  });
  return mockInfoService;
});

describe("Testing Info component", () => {
  test("it should show frontend info successfully", async () => {
    // GIVEN some frontend and backend info data are available and loaded
    const expectedFrontendInfoData: InfoProps = {
      date: "fooFrontend",
      branch: "barFrontend",
      buildNumber: "bazFrontend",
      sha: "gooFrontend",
    };
    const expectedBackendInfoData: InfoProps = {
      date: "fooBackend",
      branch: "barBackend",
      buildNumber: "bazBackend",
      sha: "gooBackend",
    };
    const infoDataPromise = Promise.resolve([expectedFrontendInfoData, expectedBackendInfoData]);

    // @ts-ignore
    InfoService.mockImplementationOnce(() => {
      return {
        loadInfo: () => {
          return infoDataPromise;
        },
      };
    });

    // WHEN the component is rendered
    render(
      <HashRouter>
        <Info />
      </HashRouter>
    );
    await act(async () => {
      await infoDataPromise;
    });

    // THEN expect no errors or warning to have occurred
    expect(console.error).not.toHaveBeenCalled();
    expect(console.warn).not.toHaveBeenCalled();
    // AND the component should be rendered
    expect(screen.getByTestId(DATA_TEST_ID.INFO_ROOT)).toBeDefined();
    expect(screen.getByTestId(DATA_TEST_ID.INFO_ROOT)).toMatchSnapshot(DATA_TEST_ID.INFO_ROOT);
    // AND the frontend info should be displayed
    expect(screen.getByTestId(DATA_TEST_ID.VERSION_FRONTEND_ROOT)).toBeDefined();
    expect(screen.getByTestId(DATA_TEST_ID.VERSION_FRONTEND_ROOT)).toMatchSnapshot(DATA_TEST_ID.VERSION_FRONTEND_ROOT);
    // AND the backend info should be displayed
    expect(screen.getByTestId(DATA_TEST_ID.VERSION_BACKEND_ROOT)).toBeDefined();
    expect(screen.getByTestId(DATA_TEST_ID.VERSION_BACKEND_ROOT)).toMatchSnapshot(DATA_TEST_ID.VERSION_BACKEND_ROOT);
  });
});
