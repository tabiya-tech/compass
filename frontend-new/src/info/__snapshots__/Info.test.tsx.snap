// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Info component it should show frontend info successfully: version-37d307ae-4f1e-4d8d-bafe-fd642f8af4dc 1`] = `
<div
  data-testid="version-37d307ae-4f1e-4d8d-bafe-fd642f8af4dc"
  style="width: 100%; height: 100%;"
>
  <div
    class="MuiBox-root css-c4dei4"
  >
    <div
      class="MuiBox-root css-1c0x74f"
      data-testid="version-frontend-37d307ae-4f1e-4d8d-bafe-fd642f8af4dc"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-10wel8w-MuiTypography-root"
      >
        Frontend
      </h6>
      <div
        class="MuiBox-root css-1w00lo4"
      >
        <div
          class="MuiBox-root css-0"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-cdd9aw-MuiTypography-root"
          >
            Date
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-7xnq80-MuiTypography-root"
          >
            fooFrontend
          </p>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-cdd9aw-MuiTypography-root"
          >
            Version
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-7xnq80-MuiTypography-root"
          >
            barFrontend
          </p>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-cdd9aw-MuiTypography-root"
          >
            Build Number
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-7xnq80-MuiTypography-root"
          >
            bazFrontend
          </p>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-cdd9aw-MuiTypography-root"
          >
            GIT SHA
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-7xnq80-MuiTypography-root"
          >
            gooFrontend
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-1c0x74f"
      data-testid="version-backend-37d307ae-4f1e-4d8d-bafe-fd642f8af4dc"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-10wel8w-MuiTypography-root"
      >
        Backend
      </h6>
      <div
        class="MuiBox-root css-1w00lo4"
      >
        <div
          class="MuiBox-root css-0"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-cdd9aw-MuiTypography-root"
          >
            Date
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-7xnq80-MuiTypography-root"
          >
            fooBackend
          </p>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-cdd9aw-MuiTypography-root"
          >
            Version
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-7xnq80-MuiTypography-root"
          >
            barBackend
          </p>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-cdd9aw-MuiTypography-root"
          >
            Build Number
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-7xnq80-MuiTypography-root"
          >
            bazBackend
          </p>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-cdd9aw-MuiTypography-root"
          >
            GIT SHA
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-7xnq80-MuiTypography-root"
          >
            gooBackend
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-opia2v"
    data-testid="feedback-button-container-31d2b110-8308-4035-90a6-519e89e7f6fa"
  >
     
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-1j5hbp5-MuiButtonBase-root-MuiButton-root"
      data-testid="feedback-button-31d2b110-8308-4035-90a6-519e89e7f6fa"
      tabindex="0"
      title="Report a bug."
      type="button"
    >
      <span
        class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-etbowg-MuiSvgIcon-root"
          data-testid="feedback-icon-31d2b110-8308-4035-90a6-519e89e7f6fa"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 8h-2.81c-.45-.78-1.07-1.45-1.82-1.96L17 4.41 15.59 3l-2.17 2.17C12.96 5.06 12.49 5 12 5c-.49 0-.96.06-1.41.17L8.41 3 7 4.41l1.62 1.63C7.88 6.55 7.26 7.22 6.81 8H4v2h2.09c-.05.33-.09.66-.09 1v1H4v2h2v1c0 .34.04.67.09 1H4v2h2.81c1.04 1.79 2.97 3 5.19 3s4.15-1.21 5.19-3H20v-2h-2.09c.05-.33.09-.66.09-1v-1h2v-2h-2v-1c0-.34-.04-.67-.09-1H20zm-6 8h-4v-2h4zm0-4h-4v-2h4z"
          />
        </svg>
      </span>
      Report a bug
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`Testing Info component it should show frontend info successfully: version-backend-37d307ae-4f1e-4d8d-bafe-fd642f8af4dc 1`] = `
<div
  class="MuiBox-root css-1c0x74f"
  data-testid="version-backend-37d307ae-4f1e-4d8d-bafe-fd642f8af4dc"
>
  <h6
    class="MuiTypography-root MuiTypography-h6 css-10wel8w-MuiTypography-root"
  >
    Backend
  </h6>
  <div
    class="MuiBox-root css-1w00lo4"
  >
    <div
      class="MuiBox-root css-0"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-cdd9aw-MuiTypography-root"
      >
        Date
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body1 css-7xnq80-MuiTypography-root"
      >
        fooBackend
      </p>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-cdd9aw-MuiTypography-root"
      >
        Version
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body1 css-7xnq80-MuiTypography-root"
      >
        barBackend
      </p>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-cdd9aw-MuiTypography-root"
      >
        Build Number
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body1 css-7xnq80-MuiTypography-root"
      >
        bazBackend
      </p>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-cdd9aw-MuiTypography-root"
      >
        GIT SHA
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body1 css-7xnq80-MuiTypography-root"
      >
        gooBackend
      </p>
    </div>
  </div>
</div>
`;

exports[`Testing Info component it should show frontend info successfully: version-frontend-37d307ae-4f1e-4d8d-bafe-fd642f8af4dc 1`] = `
<div
  class="MuiBox-root css-1c0x74f"
  data-testid="version-frontend-37d307ae-4f1e-4d8d-bafe-fd642f8af4dc"
>
  <h6
    class="MuiTypography-root MuiTypography-h6 css-10wel8w-MuiTypography-root"
  >
    Frontend
  </h6>
  <div
    class="MuiBox-root css-1w00lo4"
  >
    <div
      class="MuiBox-root css-0"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-cdd9aw-MuiTypography-root"
      >
        Date
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body1 css-7xnq80-MuiTypography-root"
      >
        fooFrontend
      </p>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-cdd9aw-MuiTypography-root"
      >
        Version
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body1 css-7xnq80-MuiTypography-root"
      >
        barFrontend
      </p>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-cdd9aw-MuiTypography-root"
      >
        Build Number
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body1 css-7xnq80-MuiTypography-root"
      >
        bazFrontend
      </p>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-cdd9aw-MuiTypography-root"
      >
        GIT SHA
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body1 css-7xnq80-MuiTypography-root"
      >
        gooFrontend
      </p>
    </div>
  </div>
</div>
`;
